packages/analytics/src/adapters/adapter.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * Primitive JavaScript type used in analytics tags.
+ */
+export type Primitive = number | string | boolean | bigint | symbol | null | undefined;
+
+/**
+ * Interface describing the requirements for a Analytics Adapter. An adapter is a tool (such as Posthog etc.) that
+ * allow the application to track event for product analysis.
+ */
+export interface AnalyticsAdapter {
+  /**
+   * Set the ID of the user currently using the application. Ignore this if you have no user system.
+   * @param id The ID of the user.
+   * @param context Optional context of the current user.
+   */
+  setUserId: (id: string, context?: Record<string, Primitive>) => void;
+  /**
+   * Set the properties/context of the user currently using the application. Ignore this if you have no user system.
+   * @param context context of the current user.
+   */
+  setUserProperties: (context: Record<string, Primitive>) => void;
+  /**
+   * Unlink any future events made on that device with that user.
+   */
+  resetUser: () => void;
+  /**
+   * Track a event.
+   * We recommend using a '[object][verb]' format for your event names, where '[object]' is the entity that the behavior relates to, and '[verb]' is the behavior itself.
+   * For example, project created, user signed up, or invite sent.
```

Comment:
Track an event.

We recommend using the...

wrap example between backticks for code formatting like `project created` etc